renderdoc/driver/vulkan/vk_resources.cpp

```diff
@@ -2919,6 +2919,94 @@ VkImageAspectFlags FormatImageAspects(VkFormat fmt)
     return VK_IMAGE_ASPECT_COLOR_BIT;
 }
 
+int ImgRefs::GetAspectCount()
+{
+  int aspectCount = 0;
+  for(auto aspectIt = ImageAspectFlagIter::begin(aspectMask);
+      aspectIt != ImageAspectFlagIter::end(); ++aspectIt)
+  {
+    ++aspectCount;
+  }
+  return aspectCount;
+}
+
+int ImgRefs::SubresourceIndex(VkImageAspectFlagBits aspect, int level, int layer)
+{
+  int aspectIndex = 0;
+  if(areAspectsSplit)
+  {
+    for(auto aspectIt = ImageAspectFlagIter::begin(aspectMask);
+        aspectIt != ImageAspectFlagIter::end(); ++aspectIt)
+    {
+      if(*aspectIt == aspect)
+        break;
+      ++aspectIndex;
+    }
+  }
+  return SubresourceIndex(aspectIndex, level, layer);
+}
+
+int ImgRefs::SubresourceIndex(int aspectIndex, int level, int layer)
+{
+  if(!areAspectsSplit)
+    aspectIndex = 0;
+  int splitLevelCount = 1;
+  if(areLevelsSplit)
+    splitLevelCount = levelCount;
+  else
+    level = 0;
+  int splitLayerCount = 1;
+  if(areLayersSplit)
+    splitLayerCount = layerCount;
+  else
+    layer = 0;
+  return (aspectIndex * splitLevelCount + level) * splitLayerCount + layer;
+}
+
+void ImgRefs::Split(bool splitAspects, bool splitLevels, bool splitLayers)
+{
+  int newSplitAspectCount = 1;
+  int oldSplitAspectCount = 1;
+  if(splitAspects || areAspectsSplit)
+  {
+    newSplitAspectCount = GetAspectCount();
+    if(splitAspects)
```

Comment:
Should this be `if(areAspectsSplit)` instead?

Reply:
Yes, good catch! Although `oldSplitAspectCount` isn't actually used, so I just deleted this.